cpp/src/parquet/column_reader.h

```diff
@@ -368,6 +368,11 @@ class PARQUET_EXPORT RecordReader {
 
   virtual void DebugPrintState() = 0;
 
+  /// \brief Returns the dictionary owned by the current decoder. Throws an
+  /// exception if the current decoder is not for dictionary encoding.
```

Comment:
Maybe add a comment that in general the only safe time to call this is before any values have been read?

Reply:
i think it's also fine to read the dictionary after reading some values to read, as long as the decoder is still valid.